polaris-react/src/components/Box/Box.tsx

```diff
@@ -207,24 +190,29 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
     },
     ref,
   ) => {
+    // eslint-disable-next-line no-nested-ternary
+    const borderStyleValue = borderStyle
+      ? borderStyle
+      : borderColor || borderWidth
+      ? 'solid'
+      : undefined;
+
+    // eslint-disable-next-line no-nested-ternary
+    const outlineStyleValue = outlineStyle
+      ? outlineStyle
+      : outlineColor || outlineWidth
+      ? 'solid'
+      : undefined;
+
     const style = {
       '--pc-box-color': color ? `var(--p-color-${color})` : undefined,
       '--pc-box-background': background
         ? `var(--p-color-${background})`
         : undefined,
-      '--pc-box-border': border ? `var(--p-border-${border})` : undefined,
-      '--pc-box-border-block-end': borderBlockEnd
-        ? `var(--p-border-${borderBlockEnd})`
-        : undefined,
-      '--pc-box-border-inline-start': borderInlineStart
-        ? `var(--p-border-${borderInlineStart})`
-        : undefined,
-      '--pc-box-border-inline-end': borderInlineEnd
-        ? `var(--p-border-${borderInlineEnd})`
-        : undefined,
-      '--pc-box-border-block-start': borderBlockStart
-        ? `var(--p-border-${borderBlockStart})`
+      '--pc-box-border-color': borderColor
+        ? `var(--p-color-${borderColor})`
```

Comment:
We don't have a `--p-color-transparent` token so this doesn't work as expected when testing a Box with transparent border like below:
```
      <Box
        background="bg"
        padding="4"
        borderWidth="5"
        borderColor="transparent"
      >
        <Icon source={PaintBrushMajor} color="base" />
      </Box>
```

Probably need to add another ternary here along the lines of below:
```suggestion
      // eslint-disable-next-line no-nested-ternary
      '--pc-box-border-color': borderColor
        ? borderColor === 'transparent'
          ? 'transparent'
          : `var(--p-color-${borderColor})`
```